uncertainties/test_uncertainties.py

```diff
@@ -2404,3 +2404,21 @@ def test_correlated_values_correlation_mat():
         assert arrays_close(
             numpy.array(cov_mat),
             numpy.array(uncert_core.covariance_matrix([x2, y2, z2])))
+
+    def test_hash():
```

Comment:
are there any NaN-likes that will have x!=x?  if you want those to be considered "matching" for e.g. dict lookups (and pandas Index lookups) those will need matching hashes

Reply:
My understanding is that there is no canonical NaN in the uncertainties package:

```
import numpy as np
import uncertainties
from uncertainties import ufloat

xx = ufloat(np.nan, 0)
yy = ufloat(np.nan, 1)

# nan+/-0
print(xx + xx)
# nan+/-2
print(yy + yy)

# False, because nan != nan
print(xx == xx)
```

The uncertainties package has unumpy.isnan which works for both scalar and arrays.  There's no dictionary compare against a nan.  The PintArray type uses unp.isnan to check whether something is a nan or not.  The pandas changes all us PintArray.isna to test NA-ness (including NaN-ness).  But if I'm missing something, please let me know!

Reply:
I've now changed the Pint-Pandas changes to use np.nan instead of pd.NA.  While I was able to get everything to work with pd.NA, things were cleaner using np.nan as the NA value for uncertain values.